src/target/mspm0.c

```diff
@@ -0,0 +1,385 @@
+#define MSPM0_FACTORYREGION_SRAMFLASH_SRAM_SZ_SHIFT      16U
+#define MSPM0_FACTORYREGION_SRAMFLASH_DATAFLASH_SZ_MASK  0xfc0000U
+#define MSPM0_FACTORYREGION_SRAMFLASH_DATAFLASH_SZ_SHIFT 26U
+
+#define MSPM0_FLASHCTL_BASE              0x400cd000U
+#define MSPM0_FLASHCTL_CMDEXEC           (MSPM0_FLASHCTL_BASE + 0x1100U)
+#define MSPM0_FLASHCTL_CMDTYPE           (MSPM0_FLASHCTL_BASE + 0x1104U)
+#define MSPM0_FLASHCTL_CMDCTL            (MSPM0_FLASHCTL_BASE + 0x1108U)
+#define MSPM0_FLASHCTL_CMDADDR           (MSPM0_FLASHCTL_BASE + 0x1120U)
+#define MSPM0_FLASHCTL_BYTEN             (MSPM0_FLASHCTL_BASE + 0x1124U)
+#define MSPM0_FLASHCTL_CMDDATAIDX        (MSPM0_FLASHCTL_BASE + 0x112cU)
+#define MSPM0_FLASHCTL_STATCMD           (MSPM0_FLASHCTL_BASE + 0x13d0U)
+#define MSPM0_FLASHCTL_CMDDATA0          (MSPM0_FLASHCTL_BASE + 0x1130U)
+#define MSPM0_FLASHCTL_CMDDATA1          (MSPM0_FLASHCTL_BASE + 0x1134U)
+#define MSPM0_FLASHCTL_CMDDATA2          (MSPM0_FLASHCTL_BASE + 0x1138U)
+#define MSPM0_FLASHCTL_CMDDATA3          (MSPM0_FLASHCTL_BASE + 0x113cU)
+#define MSPM0_FLASHCTL_CMDWEPROTA        (MSPM0_FLASHCTL_BASE + 0x11d0U)
+#define MSPM0_FLASHCTL_CMDWEPROTB        (MSPM0_FLASHCTL_BASE + 0x11d4U)
+#define MSPM0_FLASHCTL_CMDWEPROTC        (MSPM0_FLASHCTL_BASE + 0x11d8U)
+#define MSPM0_FLASHCTL_CMDWEPROTNM       (MSPM0_FLASHCTL_BASE + 0x1210U)
+#define MSPM0_FLASHCTL_CMDTYPE_NOOP      0U
+#define MSPM0_FLASHCTL_CMDTYPE_PROG      1U
+#define MSPM0_FLASHCTL_CMDTYPE_ERASE     2U
+#define MSPM0_FLASHCTL_CMDTYPE_RDVERIFY  3U
+#define MSPM0_FLASHCTL_CMDTYPE_BLVERIFY  6U
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_1WORD  (0U << 4U)
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_2WORDS (1U << 4U)
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_4WORDS (2U << 4U)
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_8WORDS (3U << 4U)
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_SECTOR (4U << 4U)
+#define MSPM0_FLASHCTL_CMDTYPE_SZ_BANK   (5U << 4U)
+#define MSPM0_FLASHCTL_CMDEXEC_EXEC      1U
+#define MSPM0_FLASHCTL_STAT_DONE         0x01U
+#define MSPM0_FLASHCTL_STAT_CMDPASS      0x02U
+
+typedef struct mspm0_flash {
+	target_flash_s target_flash;
+	uint32_t banks;
+} mspm0_flash_s;
+
+static bool mspm0_flash_erase(target_flash_s *flash, target_addr_t addr, size_t length);
+static bool mspm0_flash_write(target_flash_s *flash, target_addr_t dest, const void *src, size_t length);
+static bool mspm0_mass_erase(target_s *target);
+
+#if MSPM0_CONFIG_FLASH_DUMP_SUPPORT
+static bool mspm0_dump_factory_config(target_s *const target, const int argc, const char **const argv);
+static bool mspm0_dump_bcr_config(target_s *const target, const int argc, const char **const argv);
+
+static command_s mspm0_cmds_list[] = {
+	{"dump_factory", mspm0_dump_factory_config, "Display FACTORY registers"},
+	{"dump_bcr", mspm0_dump_bcr_config, "Display NONMAIN (BCR/BSL) registers"},
+	{NULL, NULL, NULL},
+};
+
+typedef struct conf_register {
+	uint16_t reg_offset;
+	uint16_t size_words;
+	char const *id;
```

Comment:
Please use east `const` here, `const char *` - that is how pointers are written in this code base.